collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -397,16 +380,96 @@ public void onClick(DialogInterface dialog, int i) {
         alertDialog.show();
     }
 
+    private void onElementClick(HierarchyElement element) {
```

Comment:
This method is getting called even when clicking on recycler view in `ViewFormHierarchyActivity`. The result being that had to be added to prevent the closing of the activity which can be improved. 

```
if (!(this instanceof ViewFormHierarchyActivity)) {
            finish();
        }
```

Possible solution to this: change the method visibility of `onElementClick` to `protected` and override the method in `ViewFormHierarchyActivity` with an empty body
```
@Override
    protected void onElementClick(HierarchyElement element) {
        // do nothing
    }
```
Then the check for the current instance can be removed as well